packages/devtools-evm-hardhat/src/type-extensions.ts

```diff
@@ -3,6 +3,27 @@ import { EndpointId } from '@layerzerolabs/lz-definitions'
 import { ConnectSafeConfigWithSafeAddress } from '@safe-global/protocol-kit'
 import { SimulationUserConfig } from '@/simulation/types'
 
+/**
+ * Packages containing external artifacts can be specified either
+ *
+ * - By just their package name, in which case the artifacts will be loaded from the `./artifacts` path
+ * - By their package name and a specific path to the artifacts directory
+ * - By a filesystem path
+ */
+export type ArtifactPackage = ArtifactPackageName | ArtifactPackageWithPath | ArtifactPackagePath
+
+export type ArtifactPackageName = string
+
+export interface ArtifactPackageWithPath {
+    name: ArtifactPackageName
+    path?: string
+}
+
+export interface ArtifactPackagePath {
+    name?: never
+    path: string
+}
```

Comment:
Oh this is just a way of saying:

```typescript
export interface ArtifactPackageWithPath {
    name: ArtifactPackageName
    path?: string
}

export interface ArtifactPackagePath {
    // Notice missing name attribute here
    path: string
}
```

without having to do any casting in TypeScript. If we omit the `name` property in `ArtifactPackagePath`, TypeScript would not allow us to simply do `if (artifactPackage.name == null)` since we didn't say `name` existed on the object and we would need to cast `artifactPackage` to `any` before performing the check.

The `name?: never` allows us to check for the `name` property but also ensures that the `name` property can never be set to any value.

Reply:
Ah i see, thank you for explaining!